Zinc/Private/ZincRepoIndex.m

```diff
@@ -199,22 +201,41 @@ - (void) setState:(ZincBundleState)state forBundle:(NSURL*)bundleResource
 
 - (ZincBundleState) stateForBundle:(NSURL*)bundleResource
 {
+    NSString* bundleID = bundleResource.zincBundleID;
+    ZincVersion bundleVersion = bundleResource.zincBundleVersion;
+
+    if (![self fileExistsForID:bundleID bundleVersion:bundleVersion]) {
+        return ZincBundleStateNone;
+    }
```

Comment:
Could you add a test for this specific change? I'd like to make sure we're making Zinc more maintainable with any changes we make.

Reply:
I've created a task to cover these changes with unit tests: https://elevatelabs.atlassian.net/browse/ENG-7751